jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1434,10 +1434,13 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
           capture_value = ecma_make_string_value (capture_str_p);
         }
 
+        ecma_property_descriptor_t array_item_prop_desc = ecma_make_empty_property_descriptor ();
+        array_item_prop_desc.is_value_defined = true;
+        array_item_prop_desc.value = capture_value;
         ECMA_TRY_CATCH (put_value,
-                        ecma_op_object_put (result_array_obj_p,
+                        ecma_op_object_define_own_property (result_array_obj_p,
                                             index_str_p,
-                                            capture_value,
+                                            &array_item_prop_desc,
                                             true),
```

Comment:
No, I do **NOT** wish to discard the result of property set.
RegExp.prototype.exec **should not ignore** the matched result,
but it should return matched array correctly.

And I understand that the implementation can be different with the spec.
The reason that I mentioned the spec was just to clarify the explanation.

In fact, in this case, the behavior of implementation exactly follows the spec description.
If we use `ecma_op_object_put` here,
it tries looking the prototype object to check if the property exists. (same with [[CanPut]])
And if the property exists in prototype object, (and the property is non-writable),
`ecma_op_object_put` does not create new property. 
(see jerry-core/ecma/operations/ecma-objects-general.c:334)

However, RegExp.prototype.exec must return the matched result array,
regardless of the property defined on Array prototype object. 
For example, below sample code should return the array [H,e,l,l,o]. 
And if we use `ecma_op_object_define_own_property`, it prints the correct result. 
```
Array.prototype.splice(0, 1, this);
Object.freeze(Array.prototype);
print((new String("Hello")).split(new RegExp()));
```
However if we use `ecma_op_object_put`,
it cannot add properties 'h', 'e', 'l', 'l', 'o' into result array (it just throws exception), 
so the printed result is wrong. 
(In fact, the unexpected property defined in prototype object is returned instead,
so the program get killed in random place.)